### PR TITLE
Git fixes

### PR DESCRIPTION
**Pull request by gasche**
Missing dependencies on old `git` versions.

### CODE REVIEW

**Comment by gasche** [comment]
I see a `depfail` error with the ci/datakit machines, but I don't know how to intepret it and the [Details](https://ci.ocaml.io/ocaml/opam-repository/commit/aef4403499961c824911d80ccc8d910d57578d07) page just says "No active targets for this commit". Is there an explanation somewhere of what these machines do and how to extract failure information?

**Comment by avsm** [comment]
The Datakit targets are still a little experimental -- they run a series of tests by chaining containers together across a variety of OCaml and Linux distributions, but the UI is still evolving.  In this particular case, the stable results can be clicked through now for both OPAM1.2 and 2.0.  It looks like git.1.4.0/1.4.1/1.4.2 all fail with a syntax error in a basic build (OCaml 4.03/Debian Stable), which in turn blocks other tests from running (on other OCaml versions and distros) -- hence the `depfail`.

The error is (cc @samoht)

```
- File "lib/object.mli", line 21, characters 9-13:
- Error: Syntax error
- Command exited with code 2.
```

**Comment by gasche** [comment]
I have seen this error before, it comes from a use of the `pa_with` (or however the camlp4 version of `type ... with sexplib, bin_prot, ...` was called). I hoped that adding the camlp4 dependency would fix this, but there may be another issue in either the packaging or the package's build system. Given that those are old versions of the package, this is no big deal: if I can't fix the issue after a few iterations of `opam-builder` result inspections, I will just disable them for the relevant OCaml versions.

File: packages/git/git.1.4.0/opam [hunk 1]
```diff
@@ -28,6 +28,7 @@ depends: [
   "hex"
   "conduit" {>= "0.6.0"}
   "ocamlbuild" {build}
+  "camlp4"
```
Comment (avsm) [inline packages/git/git.1.4.0/opam]: ah yes, adding `type_conv` to this list should help with the syntax errors.

Comment (gasche) [inline packages/git/git.1.4.0/opam]: Thanks, will update the PR.

Comment (gasche) [inline packages/git/git.1.4.0/opam]: Updated. Let's see if it works better this time.

Comment (avsm) [inline packages/git/git.1.4.0/opam]: Aha, this is the old `sexplib.syntax` use, so needs `sexplib {<="113.00.00"}` and `type_conv`, which will pull in the  unified extension.

**Comment by gasche** [comment]
Thanks, updated the PR again.

**Comment by gasche** [comment]
The build of git.1.4.{0,1,2} seems to now succeed, according to the CI.

**Comment by avsm** [comment]
Yes indeed, they are now failing due to cmdliner.1.0 -- thanks for all your efforts to get them this far!

**Pull request merged**